

//
// file: power.proto
// purpose: message format for interfacing the power management
// author: Tobias Simon, Ilmenau University of Technology
//


enum PowerCommand
{
   STANDING = 0; // the system is allowed to shut down main power
                 // and turn off signaling lights
   FLYING =   1; // the system must be fully powered up
}


message PowerRequest
{
   required PowerCommand command = 1;   
}


enum PowerStatus
{
   OK =       0;
   E_SYNTAX = 1;
   E_POWER =  2;
}


message PowerReply
{
   required PowerStatus status = 1;   
}


message PowerState
{
   required float voltage =   1; // main battery voltage in V
   required float current =   2; // main battery current in A
   required float capacity =  3; // battery capacity in Ah
   required float consumed =  4; // consumed capacity in Ah
   required float remaining = 5; // expected operating time in seconds
   required bool  critical =  6; // indicates low battery voltage
   optional float estimate =  7; // time estimate based on current power consumption
}

